src/PatientTestsApi/Test/Controllers/TestController.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { mock, instance, anything, verify, capture, when } from "ts-mockito";
+import { ITestDataService } from "../../Services/ITestDataService";
+import { IAuditService } from "../../Services/IAuditService";
+import { TestFixture } from "../Fixtures/TestFixture";
+import { HttpRequest } from "@azure/functions";
+import { expect } from "chai";
+import { TestController } from "../../Controllers/TestController";
+import * as HttpStatus from "http-status-codes";
+import { v4 as uuidv4 } from "uuid";
+import { BadRequestResponse } from "../../Models/BadRequestResponse";
+import { DownstreamError } from "../../Models/DownstreamError";
+import { ITest } from "../../Models/ITest";
+import { AuditingErrorResponse } from "../../Models/AuditingErrorResponse";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function createTestRequest(body: any = TestFixture.createTestForCreatingInDb()): HttpRequest {
+  const request: HttpRequest = {
+    body,
+    headers: {},
+    method: "POST",
+    url: "",
+    query: {},
+    params: { patientId: body.patientId }
+  };
+  delete request.body.patientId;
+  return request;
+}
+
+function createController(dataService?: ITestDataService, auditService?: IAuditService): TestController {
+  if (!dataService) {
+    dataService = mock<ITestDataService>();
+  }
+  if (!auditService) {
+    auditService = mock<IAuditService>();
+  }
+  return new TestController(dataService, auditService);
+}
+
+describe("TestController", async function (): Promise<void> {
+  it("Parses the test and creates it using the dataservice.", async function (): Promise<void> {
+    const dataServiceMock = mock<ITestDataService>();
+    const controller = createController(instance(dataServiceMock));
+    const request = createTestRequest();
+
+    const response = await controller.createTest(request);
+
+    verify(dataServiceMock.insertTest(anything())).once();
+    const [argument] = capture(dataServiceMock.insertTest).first();
+    expect(argument.id).is.not.null;
+    expect(response.body).is.not.null;
+    expect(response.status).is.equal(HttpStatus.CREATED);
+  });
+
```

Comment:
Maybe add a test to handle a failed insert? to see that the exception is thrown and caught correctly.

Reply:
Hmm actually that exception is not being caught, because we're not doing anything useful with it. The underlying runtime will log the exception with app Insights, and return an Http 500 error with our error's message in the body if we have debug mode on. I'm wondering if we should even catch errors to the auditing service.

Reply:
This is the same behaviour in the patient controller. Going to leave it as is for now.